Looms.PoS/Looms.PoS.Application/Mappings/ModelsResolvers/ProductModelsResolver.cs

```diff
@@ -0,0 +1,49 @@
+﻿using AutoMapper;
+using Looms.PoS.Application.Interfaces.ModelsResolvers;
+using Looms.PoS.Application.Models.Requests.Product;
+using Looms.PoS.Application.Models.Responses.Product;
+using Looms.PoS.Domain.Daos;
+
+namespace Looms.PoS.Application.Mappings.ModelsResolvers;
+
+public class ProductModelsResolver : IProductModelsResolver
+{
+    private readonly IMapper _mapper;
+
+    public ProductModelsResolver(IMapper mapper)
+    {
+        _mapper = mapper;
+    }
+
+    public ProductDao GetDaoFromRequest(CreateProductRequest createProductRequest)
+    {
+        return _mapper.Map<ProductDao>(createProductRequest);
+    }
+
+    public ProductResponse GetResponseFromDao(ProductDao productDao)
+    {
+        return _mapper.Map<ProductResponse>(productDao);
+    }
+
+    public IEnumerable<ProductResponse> GetResponseFromDao(IEnumerable<ProductDao> productDao)
+    {
+        return _mapper.Map<IEnumerable<ProductResponse>>(productDao);
+    }
+
+    ProductDao IProductModelsResolver.GetDaoFromDaoAndRequest(ProductDao originalDao, UpdateProductRequest updateProductRequest)
+    {
+        return _mapper.Map<ProductDao>(updateProductRequest) with
+        {
+            Id = originalDao.Id,
+            //TODO: idk what to do here, pls help
```

Comment:
if this is a PUT then this is okay, just make sure you copy over all properties that cannot be changed

if this is PATCH no need to map here, just return `originalDao with { }` and replace non null values from request

Reply:
fixed in "fixed indentation/layout problems, changed Variation to Variation request and made create and update requests take IEnumerable<VariationRequest>"